query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -330,11 +329,13 @@ private SearchMapping createSearchMapping(LocalQueryStatistics queryStatistics,
          numberOfShards = sharding.getShards();
       }
 
+      Integer queueSize = cache.getCacheConfiguration().indexing().writer().getQueueSize();
+      int maxConcurrency = queueSize == null ? LuceneIndexSettings.Defaults.INDEXING_QUEUE_SIZE : queueSize;
```

Comment:
1000 is OK?

Reply:
Yup, that is fine. It can go up to any number really, pending memory constraints.

Reply:
I forgot this comment!

we should consider also the queue count:

``` java
Integer queueCount = cache.getCacheConfiguration().indexing().writer().getQueueCount();
if (queueCount > 1) {
   maxConcurrency = (int) (maxConcurrency * queueCount * 0.67);
}
```

Reply:
I was debating about this. Technically it is possible for a large percentage of updates to be done to the same HS queue which would cause it to start being rejected by the HS queue. We could use the offload OperationSubmitter and immediately send it back to the FlowableProcessor. This however does introduce a different ordering and I worry about a live lock. This is the safest approach for now until we can verify those things.